google/cloud/aiplatform/training_jobs.py

```diff
@@ -602,195 +871,51 @@ def run(
         if args:
             worker_pool_spec["pythonPackageSpec"]["args"] = args
 
-        managed_model = None
+        training_task_inputs = json_format.ParseDict(
+            {
+                "workerPoolSpecs": [worker_pool_spec],
+                "baseOutputDirectory": {"output_uri_prefix": base_output_dir},
+            },
+            struct_pb2.Value(),
+        )
+
+        training_task_definition = schema.training_job.definition.custom_task
+
+        # if args need for model is incomplete
+        # TODO (b/162273530) lift requirement for predict/health route when
+        # validation lifted and move these args down
+        if model_display_name and not self._model_serving_container_image_uri:
+            raise RuntimeError(
+                """model_display_name was provided but
+                model_serving_container_image_uri was not provided when this
+                custom pipeline was constructed.
+                """
+            )
+
         # create model payload
+        container_spec = None
         if model_display_name:
-
             container_spec = gca_model.ModelContainerSpec(
                 image_uri=self._model_serving_container_image_uri,
                 predict_route=self._model_serving_container_predict_route,
                 health_route=self._model_serving_container_health_route,
             )
 
-            managed_model = gca_model.Model(
-                display_name=model_display_name, container_spec=container_spec
-            )
-
-        input_data_config = None
-        if dataset:
-            # Create fraction split spec
-            fraction_split = gca_training_pipeline.FractionSplit(
-                training_fraction=training_fraction_split,
-                validation_fraction=validation_fraction_split,
-                test_fraction=test_fraction_split,
-            )
-
-            # create input data config
-            input_data_config = gca_training_pipeline.InputDataConfig(
-                fraction_split=fraction_split,
-                dataset_id=dataset.name,
-                gcs_destination=gca_io.GcsDestination(
-                    output_uri_prefix=base_output_dir
-                ),
-            )
-
-        # create training pipeline
-        training_pipeline = gca_training_pipeline.TrainingPipeline(
-            display_name=self._display_name,
-            training_task_definition=schema.training_job.definition.custom_task,
-            training_task_inputs=json_format.ParseDict(
-                {
-                    "workerPoolSpecs": [worker_pool_spec],
-                    "baseOutputDirectory": {"output_uri_prefix": base_output_dir},
-                },
-                struct_pb2.Value(),
-            ),
-            model_to_upload=managed_model,
-            input_data_config=input_data_config,
+        managed_model = gca_model.Model(
```

Comment:
Setting this value will throw an exception if the Training script does not produce model. I think we want to previous logic where it's possible managed_model is None.

Reply:
Good catch, this was unintended.